scripts/globals/magic.lua

```diff
@@ -1950,28 +1950,25 @@ xi.magic.getCharmChance = function(charmer, target, includeCharmAffinityAndChanc
     local targetLevel     = target:getMainLvl()
     local charmres        = target:getMod(xi.mod.CHARMRES)
     local charmChance     = 50 - charmres
-    local charmerBSTLevel = 0
+    local charmerJobLevel = 0
 
     if charmer:isPC() then
-        charmerBSTLevel = charmer:getJobLevel(xi.job.BST)
-        local charmerBRDLevel = charmer:getJobLevel(xi.job.BRD)
-        -- Maiden's Virelai check
-        if charmer:getMainJob() == xi.job.BST and charmerBRDLevel > charmerBSTLevel then
-            charmerBSTLevel = charmerBRDLevel
+        if action:getID() == xi.magic.spell.MAIDENS_VIRELAI then
```

Comment:
I do not think checking like this is the best idea because spell and ability enums are different enums, so both a spell and ability relevant to charm chance could have the same value. Also getCharmChance is called from c++ code (in TryCharm function in battleutils.cpp) but using the old number of params (without the action param) so the includeCharmAffinityAndChance from that call is now being ignored.

Reply:
Updated, thank you for the input. I am still learning all of this stuff.

Reply:
> Updated, thank you for the input. I am still learning all of this stuff.

Can you change the maidensVirelai variable to be more descriptive like isMaidensVirelai